rcsbapi/data/query.py

```diff
@@ -22,29 +39,75 @@ def __init__(self, input_type: str, input_ids: Union[List[str], Dict[str, str],
             for value in input_ids.values():
                 if len(value) > input_id_limit:
                     logging.warning("More than %d input_ids. For a more readable response, reduce number of ids.", input_id_limit)
-        self._input_ids = input_ids
-        self._input_type = input_type
-        self._return_data_list = return_data_list
-        self._query = SCHEMA.construct_query(input_type=input_type, input_ids=input_ids, return_data_list=return_data_list, add_rcsb_id=add_rcsb_id)
+
         self._plural_input = False
+        """boolean indicating whether input type is plural or not (ex: "entry" vs "entries")"""
         if SCHEMA.root_dict[input_type][0]["kind"] == "LIST":
             self._plural_input = True
             if isinstance(input_ids, dict):
+                assert isinstance(input_ids, dict)  # mypy
                 self._input_ids_list: List[str] = input_ids[SCHEMA.root_dict[input_type][0]["name"]]
-            if isinstance(input_ids, list):
-                self._input_ids_list = input_ids
+
+        self._input_type = input_type
+        # automatically turn singular input_types into plural for more flexibility in number of ids
+        if self._plural_input is False:
+            plural_type = SINGULAR_TO_PLURAL[input_type]
+            if plural_type:
+                self._input_type = plural_type
+                self._plural_input = True
+
+                # if input_ids is a dict, join into PDB id format
+                if isinstance(input_ids, dict):
+                    join_id = ""
+                    for k, v in input_ids.items():
+                        assert isinstance(v, str)  # for mypy
+                        if k in ID_TO_SEPARATOR:
+                            join_id += ID_TO_SEPARATOR[k] + v
+                        else:
+                            join_id += v
+
+                    input_ids = [join_id]
+
+        self._input_ids = input_ids
+        if isinstance(self._input_ids, list):
+            self._input_ids_list = self._input_ids
+        self._return_data_list = return_data_list
+        self._query = SCHEMA.construct_query(input_type=self._input_type, input_ids=self._input_ids, return_data_list=return_data_list, add_rcsb_id=add_rcsb_id)
+        """GraphQL query as a string"""
         self._response: Optional[Dict[str, Any]] = None
 
     def get_input_ids(self) -> Union[List[str], Dict[str, List[str]], Dict[str, str]]:
```

Comment:
I went through and changed methods in query and schema not intended for users to private methods, but the `get_...` functions I left public. I was thinking that users could access the attributes through them because all the attributes are private and also access the graphql query after it was constructed (like if they wanted to copy/paste the string)